### DISCUSSION TITLE
Migration from Adafruit_Neopixel.h Lib to NeopixelBus.h -->> UpdateLength() Method

### DISCUSSION THREAD

**Discussion by JoJos1220** [Makuna/NeoPixelBus#739] [Q&A]
Hi - i am just switching my Project from the Adafruit_Neopixel.h Libary to the NeoPixelBus.h Libary because of WiFi Issues during the Adafruit Libary. Unfortunatly, i am often using the pixel.updateLength() Method because my pixel length should be adaptable during the ESP8266 runtime.

I searched through the NeoPixelBus.h lib but i was not able to find such a method - so is it just implemented?
If not, how can i update the length of the used neopixel during runtime e.g. chaniging up from 5 to 20 and back from 20 to e.g. 7?!?

Greetings!

**Comment by Makuna** [comment]
That API was added by Adafruit latter.  My suggested methods are different.  

[See this FAQ in the wiki](https://github.com/Makuna/NeoPixelBus/wiki/FAQ-%233).

**Reply by JoJos1220** [reply]
Thank you Sir!
I got it!
